File: app/src/main/java/com/wilinz/yuetingmusic/ui/user/UserListAdapter.java
package com.wilinz.yuetingmusic.ui.user;

import android.annotation.SuppressLint;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.wilinz.yuetingmusic.R;
import com.wilinz.yuetingmusic.data.model.User;
import com.wilinz.yuetingmusic.databinding.ItemUserBinding;

import java.util.List;

public class UserListAdapter extends RecyclerView.Adapter<UserListAdapter.UserListViewHolder> {

    private List<User> users;
    private OnItemClickListener listener;

    public interface OnItemClickListener {
        void onItemClick(List<User> users, int index, User user);
    }

    public UserListAdapter(@NonNull List<User> users) {
        this.users = users;
    }

    public void setUsers(List<User> users) {
        this.users = users;
        notifyDataSetChanged();
    }

    public void setOnItemClickListener(OnItemClickListener l) {
        listener = l;
    }

    static class UserListViewHolder extends RecyclerView.ViewHolder {

        ItemUserBinding binding;

        public UserListViewHolder(@NonNull ItemUserBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }

    @NonNull
    @Override
    public UserListViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemUserBinding binding = ItemUserBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        UserListViewHolder holder = new UserListViewHolder(binding);
        binding.getRoot().setOnClickListener(v -> {
            if (listener != null) {
                int index = holder.getAbsoluteAdapterPosition();
                listener.onItemClick(users, index, users.get(index));
            }
        });
        return holder;
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(@NonNull UserListViewHolder holder, int position) {
        ItemUserBinding binding = holder.binding;
        User user = users.get(position);
        binding.name.setText(user.username);
//        binding.serialNumber.setText((position + 1) + "");
        binding.secondName.setText(user.nickname);
        if (user.isActive){
            binding.tag.setText("当前用户");
        }else {
            binding.tag.setText("");
        }
        Glide.with(binding.songAvatar)
                .load(user.avatar)
                .error(R.drawable.avatar2)
                .into(binding.songAvatar);
    }

    @Override
    public int getItemCount() {
        return users.size();
    }


}

File: app/src/main/java/com/wilinz/yuetingmusic/ui/user/UserListAdapter.kt
package com.wilinz.yuetingmusic.ui.user

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.wilinz.yuetingmusic.R
import com.wilinz.yuetingmusic.data.model.User
import com.wilinz.yuetingmusic.databinding.ItemUserBinding
import com.wilinz.yuetingmusic.ui.user.UserListAdapter.UserListViewHolder
import com.wilinz.yuetingmusic.util.ScreenUtil

class UserListAdapter(private var users: List<User>) : RecyclerView.Adapter<UserListViewHolder>() {
    private var listener: ((users: List<User>, index: Int, user: User)->Unit)? = null

    interface OnItemClickListener {
        fun onItemClick(users: List<User>?, index: Int, user: User?)
    }

    fun setUsers(users: List<User>) {
        this.users = users
        notifyDataSetChanged()
    }

    fun setOnItemClickListener(l: (users: List<User>, index: Int, user: User)->Unit) {
        listener = l
    }

    class UserListViewHolder(var binding: ItemUserBinding) : RecyclerView.ViewHolder(
        binding.root
    )

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserListViewHolder {
        val binding = ItemUserBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        val holder = UserListViewHolder(binding)
        binding.root.setOnClickListener { v: View? ->
            val currentUserIndex = users.indexOfFirst { user: User -> user.isActive }
            val index = holder.absoluteAdapterPosition
            if (index != currentUserIndex && listener != null) {
                listener?.invoke(users, index, users[index])
            }
        }
        return holder
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: UserListViewHolder, position: Int) {
        val binding = holder.binding
        val user = users[position]
        binding.name.text = user.username
        //        binding.serialNumber.setText((position + 1) + "");
        binding.secondName.text = user.nickname
        if (user.isActive) {
            binding.tag.text = "当前用户"
        } else {
            binding.tag.text = ""
        }
        val size = ScreenUtil.dpToPx(binding.songAvatar.context, 48)
        Glide.with(binding.songAvatar)
            .load(user.avatar + "?param=${size}y${size}")
            .error(R.drawable.avatar2)
            .into(binding.songAvatar)
    }

    override fun getItemCount(): Int {
        return users.size
    }
}